Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -0,0 +1,356 @@
+    % % expected volumes per repetition (for the amount of x.Q.EchoTimes, PLDs, and labeling durations)
+    % if nRepetitions~=round(nRepetitions)
+    %     error(['nVolumes ' xASL_num2str(size(PWI4D,4)) ' cannot be composed of ' xASL_num2str(nVolumesPerRepetition) ' volumes per repetition']);
+    % end
+
+
+
+    
+    
+% =====================================================================
+% C) Single- and multi-PLD subtraction
+elseif bCreatePWI4D
+    %% 4. Create PWI4D
+    if nVolumes>1 && ~x.modules.asl.bContainsDeltaM
+        % Paired subtraction
+        [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASL4D);
+        PWI4D = ControlIm - LabelIm;
+        
+        % Skip every other value in the vectors as they were stored for both control and label images 
+        x.Q.EchoTime_PWI4D = x.Q.EchoTime(1:2:end);
+        x.Q.InitialPLD_PWI4D = x.Q.Initial_PLD(1:2:end);
+        x.Q.LabelingDuration_PWI4D = x.Q.LabelingDuration(1:2:end);
+    else % the same but then without subtraction
+        PWI4D = ASL4D;
+    end
+
+end
+   
+if bCreatePWI3D
+    %% 5. Create PWI3D
+    
+    % After averaging across PLDs, we'll obtain these unique PLDs+LD combinations
+    % indexAverage_PLD_LabDur lists for each original position to where it should be averaged
+    [~, ~, iUnique_TE_PLD_LabDur] = unique([x.Q.EchoTime_PWI4D(:), x.Q.InitialPLD_PWI4D(:), x.Q.LabelingDuration_PWI4D(:)], 'stable', 'rows');
+    
+    % MultiPLD-multiLabDur PWI3D after averaging
+    for iTE_PLD_LabDur = 1:max(iUnique_TE_PLD_LabDur)
+        indicesAre = iUnique_TE_PLD_LabDur == iTE_PLD_LabDur;
+        PWI3D(:, :, :, iTE_PLD_LabDur) = xASL_stat_MeanNan(PWI4D(:, :, :, indicesAre), 4); % Averaged PWI4D
+        x.Q.EchoTime_PWI3D(iTE_PLD_LabDur) = xASL_stat_MeanNan(x.Q.EchoTime_PWI4D(indicesAre));
+        x.Q.InitialPLD_PWI3D(iTE_PLD_LabDur) = xASL_stat_MeanNan(x.Q.InitialPLD_PWI4D(indicesAre));
+        x.Q.LabelingDuration_PWI3D(iTE_PLD_LabDur) = xASL_stat_MeanNan(x.Q.LabelingDuration_PWI4D(indicesAre));
+    end
+end
+
+%% 6. Create PWI
+% We create a dummy CBF image for registration purposes
+% The earliest echo, the latest PLD and the longest labeling duration
+% are the best for this, having most SNR, CBF-weighting, and SNR,
+% respectively. 
+% But from a single volume, we have little signal.
+% So we take a weighted average accordingly
+
+%% This would be taking a single volume only
+% % The earliest echo has the most SNR for perfusion-weighting
+% iMinTE = x.Q.EchoTime_PWI3D == min(x.Q.EchoTime_PWI3D(:));
+% % The latest PLD has the most perfusion-weighting
+% iMaxPLD = x.Q.InitialPLD_PWI3D == max(x.Q.InitialPLD_PWI3D(:));
+% % The longest labeling duration has the most SNR and most perfusion-weighting
+% iMaxLabelingDuration = x.Q.LabelingDuration_PWI3D == max(x.Q.LabelingDuration_PWI3D(:));
+% % We take the index that has all these        
+% i3D = iMinTE & iMaxPLD & iMaxLabelingDuration;
```

Comment:
This might not exist -> see above.

Reply:
For this we still need to reach consensus, as we will create a weighting matrix, see discussion in #997. But you can resolve this now, because in the code we now simply take the mean of all PLD and LD combinations (shortest TE only).